Rakefile

```diff
@@ -86,6 +87,25 @@ namespace :gems do
   end
 end
 
+class Hash
+  def sort_by_key(recursive = false, &block)
+    keys.sort(&block).each_with_object({}) do |key, seed|
+      seed[key] = self[key]
+      seed[key] = seed[key].sort_by_key(true, &block) if recursive && seed[key].is_a?(Hash)
+      seed
+    end
+  end
+end
```

Comment:
I have never used Ruby's refinements feature but this might actually fit it's intended use case. https://docs.ruby-lang.org/en/2.4.0/syntax/refinements_rdoc.html. Also, I would like to add... https://www.youtube.com/watch?v=plzw1NWVYSM&t=8s

Reply:
Disclosure: I typed "rubocop.yml sort" into google and built this from https://gist.github.com/ream88/e510ca628d35515e5cbbcc18e2ea3238 _ream88 is the real MVP!

Being critical: I think `&block` could be trimmed. I don't mind [cargo-culting](https://en.wikipedia.org/wiki/Cargo_cult_programming) it along, but it's not needed.